thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/PresenterSavior.java

```diff
@@ -125,23 +140,49 @@ public void onActivityFinished(final Activity activity, final String activityId)
 
     @Override
     @Nullable
-    public TiPresenter recover(final String presenterId, @NonNull final Activity activity) {
-        final ActivityScopedPresenters scope = getScope(activity);
+    public TiPresenter recover(final String presenterId, @NonNull final Object host) {
+        final PresenterScope scope = getScope(host);
         if (scope == null) {
             return null;
         }
         return scope.get(presenterId);
     }
 
     @Override
-    public String save(@NonNull final TiPresenter presenter, @NonNull final Activity activity) {
-        final String id = generatePresenterId(presenter);
+    public String save(@NonNull final TiPresenter presenter, @NonNull final Object host) {
+
+        // the hostId
+        String hostId = null;
+
+        PresenterScope scope = getScope(host);
+        if (scope == null) {
+            // create a new scope
+            scope = new PresenterScope();
+            hostId = generateHostId(host);
+            mScopes.put(hostId, scope);
+        }
+        final String presenterId = generatePresenterId(presenter);
+        scope.save(presenterId, presenter);
+
+        // register host observer
+        if (host instanceof Activity) {
+            final Activity activity = (Activity) host;
+            observeActivityFinish(activity, hostId);
+        } else {
+            // currently only Activity is supported as host
+            throw new IllegalStateException(
+                    "Host has unknown type " + host.getClass().getSimpleName()
+                            + " and is not supported.");
+        }
 
-        final ActivityScopedPresenters scope = getScopeOrCreate(activity);
-        scope.save(id, presenter);
         printRemainingPresenter();
 
-        return id;
+        return presenterId;
+    }
+
+    private String generateHostId(final Object host) {
+        return host.getClass().getSimpleName()
+                + ":" + System.nanoTime();
```

Comment:
Don't we want to add the hashCode as well? 🤔 
Like in the `generatePresenterId`